.github/workflows/wp-media.yml

```diff
@@ -0,0 +1,191 @@
+---
+name: Media AMI
+
+permissions:
+  id-token: write
+  contents: write
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploy-environment:
+        type: choice
+        description: Environment
+        required: true
+        options:
+        - dev
+        - staging
+        - live
+
+jobs:
+# ------------------------------------------------------------------------------
+# dev deployment
+# ------------------------------------------------------------------------------
+  media-ami-deployment-dev:
+    if: github.event.inputs.deploy-environment == 'dev'
+    runs-on: ubuntu-latest
+    environment: media-dev
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v2
+        with:
+          ref: "${{ github.event.inputs.branch }}"
+
+      - name: set ansible var file for dev
+        run: |
+          echo "ANSIBLE_VAR_FILE=vars_dev.yml" >> $GITHUB_ENV
+          echo "SSH_KEY_FILE=media-app-dev-eu-west-2" >> $GITHUB_ENV
+
+      - name: aws variable settings
+        run: |
+          echo AWS_ROLE_ARN=${{ secrets.AWS_ROLE_ARN }} >> $GITHUB_ENV
+          echo AWS_DEFAULT_REGION=eu-west-2 >> $GITHUB_ENV
+          echo AWS_ACCESS_KEY_ID=${{ secrets.AWS_ACCESS_KEY_ID }} >> $GITHUB_ENV
+          echo AWS_SECRET_ACCESS_KEY=${{ secrets.AWS_SECRET_ACCESS_KEY }} >> $GITHUB_ENV
+
+      - name: write ssh key file
+        run: |
+          echo "${{ secrets.SSH_KEY }}" | tr -d '\r' > ${{ env.SSH_KEY_FILE }}
+          chmod 0600 ${{ env.SSH_KEY_FILE }}
+
+      - name: install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3
+      - name: upgrade pip
+        run: pip install --upgrade pip
+      - name: install boto3
+        run: pip3 install boto3 requests pyjwt
+      - name: install wheel
+        run: pip3 install wheel requests pyjwt
```

Comment:
Aren't the "requests pyjwt" module references redundant here?

Reply:
I'm not a python specialist. We could try to remove the requests and see if it still works.